pkg/reporting/app_airgap.go

```diff
@@ -1,28 +1,147 @@
 package reporting
 
 import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
 	"github.com/pkg/errors"
-	"github.com/replicatedhq/kots/pkg/store"
+	"github.com/replicatedhq/kots/pkg/api/reporting/types"
+	"github.com/replicatedhq/kots/pkg/k8sutil"
+	"github.com/replicatedhq/kots/pkg/logger"
+	"github.com/replicatedhq/kots/pkg/util"
+	kuberneteserrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	InstanceReportSecretNameFormat = "kotsadm-%s-instance-report"
+	InstanceReportSecretKey        = "report"
+	InstanceReportEventLimit       = 4000
 )
 
+var instanceReportMtx = sync.Mutex{}
+
 func (r *AirgapReporter) SubmitAppInfo(appID string) error {
-	a, err := store.GetStore().GetApp(appID)
+	a, err := r.store.GetApp(appID)
 	if err != nil {
-		if store.GetStore().IsNotFound(err) {
+		if r.store.IsNotFound(err) {
 			return nil
 		}
-		return errors.Wrap(err, "failed to get airgaped app")
+		return errors.Wrap(err, "failed to get airgapped app")
 	}
 
-	license, err := store.GetStore().GetLatestLicenseForApp(a.ID)
+	license, err := r.store.GetLatestLicenseForApp(a.ID)
 	if err != nil {
 		return errors.Wrap(err, "failed to get license for airgapped app")
 	}
 	reportingInfo := GetReportingInfo(appID)
 
-	err = store.GetStore().SaveReportingInfo(license.Spec.LicenseID, reportingInfo)
+	event := GetInstanceReportEvent(license.Spec.LicenseID, reportingInfo)
+
+	if err := CreateInstanceReportEvent(r.clientset, util.PodNamespace, a.Slug, event); err != nil {
+		return errors.Wrap(err, "failed to create instance report event")
+	}
+
+	return nil
+}
+
+func GetInstanceReportEvent(licenseID string, reportingInfo *types.ReportingInfo) InstanceReportEvent {
+	// not using the "cursor" packages because it doesn't provide access to the underlying int64
+	downstreamSequence, err := strconv.ParseUint(reportingInfo.Downstream.Cursor, 10, 64)
+	if err != nil {
+		logger.Debugf("failed to parse downstream cursor %q: %v", reportingInfo.Downstream.Cursor, err)
```

Comment:
This maintains existing behavior by not bailing if we fail to parse the downstream sequence. Any reason to change this?

Reply:
nope